### DISCUSSION TITLE
Is JSONSchema the best way to handle expressing schemas?

### DISCUSSION THREAD

**Discussion by sherifnada** [airbytehq/airbyte#2395] [Connector Ideas and Features]
Braindumping. Feel free to add to this. 

Some nice things about it: 
1. big ecosystem around it
2. Reasonably expressive

Some issues it presents: 
1. It does not have sets
2. it does not allow specifying precision on numeric types

**Comment by michel-tricot** [comment]
for the numeric types, isn't it something you can do with formats?

**Reply by cgardens** [reply]
@michel-tricot I thought the same as you, but apparently it's not officially part of the spec. I do think open api does use format to specify ints versus longs, but i think that's a dialect.

**Comment by michel-tricot** [comment]
For the `set` piece: https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.6.4.3
